### PR TITLE
feat(route): Tapechat 动态

### PR DESCRIPTION
**Pull request by ch1y4**
## 该 PR 相关 Issue / Involved issue

无

## 完整路由地址 / Example for the proposed route(s)

```routes
/tapechat/dynamic/b006c523a261f0c366f94714c1c8f1d1
```

## 新 RSS 检查列表 / New RSS Script Checklist

- [x] 新的路由 New Route
  - [ ] 跟随 [[v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html)](https://docs.rsshub.app/joinus/script-standard.html) Follows [[v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [[日期和时间](https://docs.rsshub.app/joinus/pub-date.html)](https://docs.rsshub.app/joinus/pub-date.html) [[date and time](https://docs.rsshub.app/en/joinus/pub-date.html)](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

### CODE REVIEW

**Review by Rongronggg9 — CHANGES_REQUESTED** [review]
A new route should follow the v2 standard. Otherwise, it would probably not be accepted.

Your master branch is not clean, please close this PR, create a new branch based on the **upstream** master then cherry-pick your commits. Afterward, create a new PR from that branch.
Of course, you can do it on your master branch in situ if you know how to. In this way, you don't need to create a new PR. But it is not recommended to create a PR from the master branch.

I saw a lot of HTML jointing actions in your route. In v2 standard, you should use art-template to finish this work.

File: lib/pr.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<!--
```
Comment (Rongronggg9) [inline lib/pr.md:1]: Please do not copy and/or modify the PR template. You just need to fill in it while creating a PR on github.

File: lib/routes/tapechat/tapechat.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (item.item.hasOwnProperty('imgList')) {
+                for (let i = 0; i < item.item.imgList.length; i++) {
+                    const v = JSON.stringify(item.item.imgList[i]);
+                    const x = v.substring(10);
+                    images.push(`<img src=${x}>`);
+                }
+            }
+            return {
+                title: '',
+                description: `<meta name="referrer" content="no-referrer">${item.item.txtContent}<br>${images.join('')}`,
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:43]:
```suggestion
                description: `${item.item.txtContent}<br>${images.join('')}`,
```
`description` is not a web page, `<meta>` tags won't work.
You don't need to apply a no-referer policy on images, RSSHub will do it automatically.

File: lib/routes/tapechat/tapechat.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        link: `https://www.tapechat.net/profileMb.html?user=${userId}`,
+        description: String(description),
+        image: avatar,
+        item: data.map((item) => {
+            const images = [];
+            if (item.item.hasOwnProperty('imgList')) {
+                for (let i = 0; i < item.item.imgList.length; i++) {
+                    const v = JSON.stringify(item.item.imgList[i]);
+                    const x = v.substring(10);
+                    images.push(`<img src=${x}>`);
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:38]:
```suggestion
                    images.push(`<img src="${x}">`);
```

File: lib/routes/tapechat/tapechat.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        headers: {
+            Referer: `https://www.tapechat.net/profileMb.html?user=${userId}`,
+            Authorization: String(token),
+        },
+    });
+    const data = getDynamicFromUser.data.content.data;
+    ctx.state.data = {
+        title: `${nickname} 的 Tape 动态`,
+        link: `https://www.tapechat.net/profileMb.html?user=${userId}`,
+        description: String(description),
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:30]: Why force type conversion?

File: lib/routes/tapechat/tapechat.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    });
+    const nickname = getUserInfo.data.content.nickName;
+    const avatar = getUserInfo.data.content.customPic;
+    const description = getUserInfo.data.content.selfDescription;
+    const getDynamicFromUser = await got({
+        method: 'get',
+        url: `https://apiv4.tapechat.net/dynamic/userdynamic/${userId}?pageSize=${pageSize}`,
+        headers: {
+            Referer: `https://www.tapechat.net/profileMb.html?user=${userId}`,
+            Authorization: String(token),
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:23]: Why force type conversion?

File: lib/routes/tapechat/tapechat.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        title: `${nickname} 的 Tape 动态`,
+        link: `https://www.tapechat.net/profileMb.html?user=${userId}`,
+        description: String(description),
+        image: avatar,
+        item: data.map((item) => {
+            const images = [];
+            if (item.item.hasOwnProperty('imgList')) {
+                for (let i = 0; i < item.item.imgList.length; i++) {
+                    const v = JSON.stringify(item.item.imgList[i]);
+                    const x = v.substring(10);
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:37]: What's this?

File: lib/routes/tapechat/tapechat.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                    images.push(`<img src=${x}>`);
+                }
+            }
+            return {
+                title: '',
```
Comment (Rongronggg9) [inline lib/routes/tapechat/tapechat.js:42]:
```suggestion
                title: item.item.txtContent,
```

**Review by Rongronggg9 — CHANGES_REQUESTED** [review]

**Comment by DIYgod** [comment]
pr 时间过长，将先被关闭，如果还有疑问可以随时在下面继续讨论

**Pull request closed without merging**